### PR TITLE
Restore GetAwaiter behavior of TaskBlockingExpressionVisitor.

### PR DESCRIPTION
**Pull request by anpete**
And corresponding changes to task lifting and Include compiler.

### CODE REVIEW

File: src/EFCore/Query/ExpressionVisitors/Internal/TaskLiftingExpressionVisitor.cs [hunk 1]
```diff
@@ -150,6 +151,38 @@ protected override Expression VisitMember(MemberExpression memberExpression)
...
+                _taskExpressions.Add(
+                    Expression.Lambda<Func<Task<object>>>(
+                        Expression.Call(
+                            _toObjectTask.MakeGenericMethod(
+                                methodCallExpression.Method.ReturnType),
+                            methodCallExpression.Arguments[0])));
+
+                if (CancellationTokenParameter == null)
+                {
+                    Visit(methodCallExpression.Arguments[0]);
```
Comment (smitpatel) [inline src/EFCore/Query/ExpressionVisitors/Internal/TaskLiftingExpressionVisitor.cs]: Why do we need to visit only in some cases?

**Comment by smitpatel** [comment]
No tests?

**Comment by anpete** [comment]
@smitpatel Used heavily by existing tests.

File: src/EFCore/Query/ExpressionVisitors/Internal/TaskLiftingExpressionVisitor.cs (hunk 1 above)
Comment (anpete) [reply src/EFCore/Query/ExpressionVisitors/Internal/TaskLiftingExpressionVisitor.cs]: Existing code but I think we use it to signal when to stop.

**Pull request merged**